common/hash_data_other.go

```diff
@@ -0,0 +1,70 @@
+//go:build !windows
+// +build !windows
+
+package common
+
+import (
+	"encoding/json"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+)
+
+func TryGetHashData(fullpath string) (SyncHashData, error) {
+	// get meta file name
+	dir, fn := filepath.Split(fullpath)
+	var metaFile string
+	if dir == "." { // Hide the file on UNIX-like systems (e.g. Linux, OSX)
+		metaFile = "./." + fn + AzCopyHashDataStream
+	} else {
+		metaFile = dir + "/." + fn + AzCopyHashDataStream
+	}
+	// open file for reading
+	f, err := os.OpenFile(metaFile, os.O_RDONLY, 0644)
+	if err != nil {
+		return SyncHashData{}, err
+	}
+	defer f.Close()
+
+	buf, err := io.ReadAll(f)
+	if err != nil {
+		return SyncHashData{}, err
+	}
+
+	var out SyncHashData
+	err = json.Unmarshal(buf, &out)
+
+	return out, err
+}
+
+func PutHashData(fullpath string, data SyncHashData) error {
+	// get meta file name
+	dir, fn := filepath.Split(fullpath)
+	var metaFile string
+	if dir == "." { // Hide the file on UNIX-like systems (e.g. Linux, OSX)
+		metaFile = "./." + fn + AzCopyHashDataStream
+	} else {
+		metaFile = dir + "/." + fn + AzCopyHashDataStream
+	}
+	// open file for writing; truncate.
+	f, err := os.OpenFile(metaFile, os.O_CREATE|os.O_TRUNC|os.O_RDWR, 0644)
```

Comment:
We are saving the hash data in a separate file. Can we not save it in extended attributes of the source file itself.

Reply:
I opted to write to a separate file as an OS-agnostic solution. I can look into a Linux-specific implementation to use xattr.

Reply:
Should we want to use another open-source dependency, https://github.com/pkg/xattr provides an OS-agnostic implementation for most *NIX based operating systems, but more importantly, the two we officially support. The package utilizes the BSD 2-clause "Simplified" license, which, we have pulled other packages in under.

That would be a quick & dirty way to implement that.

Reply:
We could have a detection mechanism, but that's a lot of work for minimal gain, and when a transfer can easily span multiple filesystems, we *have* to accept that sometimes, we're just not going to be able to write the xattr, and that HBS is going to come with a tremendous performance caveat every time.